notebooks/using-mlflow/logging-models/logging_model_with_mlflow.ipynb

```diff
@@ -0,0 +1,600 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1e1037fd",
+   "metadata": {},
+   "source": [
+    "Some times, autolog can't guess the best way to log a model and hence you would need to do it yourself. For those cases, you can indicate autolog to log everything but the model. You do that using the parameter `log_models=False` of `autolog()` function."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0a3ba624",
+   "metadata": {},
+   "source": [
+    "### Logging models supported by MLFlow\n",
+    "\n",
+    "If you need to log the models in a particular way, then you can use the method `log_model` to log the models as you need to. Usually, you will log the model in this way when:\n",
+    "\n",
+    "* You need to indicate `pip` packages or dependencies different from the ones that are automatically detected.\n",
+    "* You need to indicate a `conda` environment different from the default one.\n",
+    "* Your models uses a signature different from the one inferred. This is specifically important when you deal with inputs that are tensors where the signature needs specific shapes.\n",
+    "* You want to include input examples.\n",
+    "* You want to include specific artifacts into the package that will be needed.\n",
+    "* Somehow the default behaviour of autolog doesn't fill your purpoise.\n",
+    "\n",
+    "To log a model, you use the `log_method` model of the flavor you are working with. For instance, the following code logs a model for an XGBoost classifier:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ba4faceb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import mlflow\n",
+    "from xgboost import XGBClassifier\n",
+    "from sklearn.metrics import accuracy_score\n",
+    "from mlflow.models import infer_signature\n",
+    "\n",
+    "with mlflow.start_run():\n",
+    "\n",
+    "    mlflow.xgboost.autolog(log_models=False)\n",
+    "\n",
+    "    model = XGBClassifier(use_label_encoder=False, eval_metric=\"logloss\")\n",
+    "    model.fit(X_train, y_train, eval_set=[(X_test, y_test)], verbose=False)\n",
+    "    y_pred = model.predict(X_test)\n",
+    "\n",
+    "    accuracy = accuracy_score(y_test, y_pred)\n",
+    "    mlflow.log_metric(\"accuracy\", accuracy)\n",
+    "\n",
+    "    signature = infer_signature(X_test, y_test)\n",
+    "    mlflow.xgboost.log_model(model, \"classifier\", signature=signature)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3bfad4c4",
+   "metadata": {},
+   "source": [
+    "If you need to indicate a custom environment with packages, you can use:\n",
+    "\n",
+    "```python\n",
+    "from mlflow.utils.environment import _mlflow_conda_env\n",
```

Comment:
I know, but I find it a bit more handy rather than creating the entire schema for the environment definition file by hand. Do you have another way to do this without writing the file manually?